### PR TITLE
LookupResolver: Implement LookupIP() and LookupIPAddr()

### PR DESCRIPTION
**Pull request by amery**
and simplify `msgToNewIPq()`

### CODE REVIEW

File: resolver_netip.go [hunk 1]
```diff
@@ -2,13 +2,44 @@ package resolver
...
 	"context"
+	"net"
 	"net/netip"
 	"sync"
 
 	"darvaza.org/core"
 	"github.com/miekg/dns"
 )
 
+// LookupIPAddr returns the IP addresses of a host
```
Comment (karasz) [inline resolver_netip.go:13]:
```go
// LookupIPAddr returns the IP address of a host
// in the form of a slice of net.IPAddr
```

File: resolver_netip.go [hunk 2]
```diff
@@ -2,13 +2,44 @@ package resolver
...
+	out := make([]net.IPAddr, 0, len(addrs))
+
+	for _, ip := range addrs {
+		out = append(out, net.IPAddr{IP: ip})
+	}
+
+	return out, err
+}
+
+// LookupIP returns the IP addresses of a host
```
Comment (karasz) [inline resolver_netip.go:27]:
```go
// LookupIPAddr returns the IP address of a host
// in the form of a slice of net.IP
```

File: resolver_netip.go [hunk 3]
```diff
@@ -120,41 +151,28 @@ func eqNetIP(ip1, ip2 netip.Addr) bool {
...
 					s = append(s, ip)
 				}
-			}
+			})
 		}
-	}
-	return s, nil
-}
 
-func validMsg(m *dns.Msg) bool {
```
Comment (karasz) [inline resolver_netip.go:148]: Better than removing it, maybe a better approach is not move it centrally and make it callable for any `*dns.Msg`

**Review by karasz — COMMENTED** [review]
See comments

**Comment by amery** [comment]
@karasz amended

**Review by karasz — APPROVED** [review]
LGTM

**Comment by amery** [comment]
`v0.4.1` tagged

**Pull request merged**